### PR TITLE
Fix various spelling corrections

### PR DESCRIPTION
**Pull request by luzpaz**

### CODE REVIEW

File: codespell_lib/data/dictionary.txt [hunk 1]
```diff
@@ -9942,6 +9954,9 @@ degreeees->degrees
 degreees->degrees
 degres->degrees, digress,
 degress->degrees, digress,
+deifne->define
+deifned->defined
+deifnes->defines
```
Comment (vikivivi) [inline codespell_lib/data/dictionary.txt:9959]:
```text
deifning->defining

dedfined->defined
definde->define, defined, defines,
definding->defining
```
* https://github.com/search?q=deifning&type=code  240
* https://github.com/search?q=dedfined&type=code 239
* https://github.com/search?q=definde&type=code 10922
* https://github.com/search?q=definding&type=code 1557

File: codespell_lib/data/dictionary.txt [hunk 2]
```diff
@@ -10058,6 +10073,7 @@ delivative->derivative
 delivatives->derivatives
 deliverate->deliberate
 delivermode->deliverymode
+deliverying->delivering
```
Comment (vikivivi) [inline codespell_lib/data/dictionary.txt:10076]:
```text
delibery->delivery
deliverys->deliveries, delivers,
```
* https://github.com/search?q=delibery&type=code 1877
* https://github.com/search?q=deliverys&type=code 15584

File: codespell_lib/data/dictionary.txt [hunk 3]
```diff
@@ -21242,6 +21279,7 @@ miminal->minimal
 miminally->minimally
 miminaly->minimally
 miminize->minimize
+miminized->minimized
```
Comment (vikivivi) [inline codespell_lib/data/dictionary.txt:21282]:
```text
miminizes->minimizes
miminizing->minimizing

miminise->minimise
miminised->minimised
miminises->minimises
miminising->minimising
```
* https://github.com/search?q=miminizes&type=code  515
* https://github.com/search?q=miminizing&type=code  366
* https://github.com/search?q=miminise&type=code  167
* https://github.com/search?q=miminised&type=code  1253
* https://github.com/search?q=miminises&type=code  6
* https://github.com/search?q=miminising&type=code  9

File: codespell_lib/data/dictionary.txt [hunk 4]
```diff
@@ -29093,6 +29144,10 @@ robustnes->robustness
 Rockerfeller->Rockefeller
 rococco->rococo
 rocord->record
+rocorded->recorded
+rocording->recording
+rocordings->recordings
+rocords->records
```
Comment (vikivivi) [inline codespell_lib/data/dictionary.txt:29150]:
```text
rocorder->recorder
```
https://github.com/search?q=rocorder&type=code  259

File: codespell_lib/data/dictionary.txt [hunk 5]
```diff
@@ -30908,6 +30968,8 @@ specfying->specifying
 speciafied->specified
 specialiced->specialised, specialized,
 specialitzed->specialised, specialized,
+specializaiton->specialization
+specializaitons->specializations
```
Comment (vikivivi) [inline codespell_lib/data/dictionary.txt:30972]:
```text
specialisaiton->specialisation
specialisaitons->specialisations
```
* https://github.com/search?q=specialisaiton&type=code  21
* https://github.com/search?q=specialisaitons&type=code  2

File: codespell_lib/data/dictionary.txt [hunk 6]
```diff
@@ -33369,6 +33435,7 @@ threadsave->threadsafe
 threah->thread, threat,
 threashold->threshold
 threasholds->thresholds
+threated->threatened, treated,
```
Comment (vikivivi) [inline codespell_lib/data/dictionary.txt:33438]: Please suggest `threaded` as first choice.
```text
threated->threaded, threatened, treated,
```
* https://github.com/search?q=threated&type=code  78766

File: codespell_lib/data/dictionary.txt [hunk 7]
```diff
@@ -35055,7 +35124,7 @@ unsiged->unsigned
 unsigend->unsigned
 unsignd->unsigned
 unsignificant->insignificant
-unsing->using
+unsing->using, unsign,
```
Comment (vikivivi) [inline codespell_lib/data/dictionary.txt:35127]:
```text
unsing->using, unsigned,
```
`unsign` is not a valid word in various dictionaries (aspell, hunspell, ispell, aoo-mozilla-en-dict, browser).
Please suggest `unsigned`. If you type `unsign integer` in Google search box before hitting ENTER key,
you will notice Google suggest you with `unsigned integer` for your search

File: codespell_lib/data/dictionary.txt [hunk 8]
```diff
@@ -36183,6 +36252,7 @@ wasnt'->wasn't
 wasnt->wasn't
 wasnt;->wasn't
 wass->was
+wastefull->wasteful
```
Comment (vikivivi) [inline codespell_lib/data/dictionary.txt:36255]: Please suggest `wastefully` as well.
```text
wastefull->wasteful, wastefully,
wastefullness->wastefulness
```
* https://github.com/search?q=wastefull&type=code  5168
* https://github.com/search?q=wastefullness&type=code  97

**Review by vikivivi — CHANGES_REQUESTED** [review]

File: codespell_lib/data/dictionary.txt [hunk 9]
```diff
@@ -28149,6 +28198,7 @@ repackged->repackaged
 repaitnt->repaint
 repant->repaint, repent,
 repants->repaints, repents,
+reparamterization->reparameterization
```
[the remark this answers is not in the document]
Comment (luzpaz) [reply codespell_lib/data/dictionary.txt:28201]: Do you suggest to move to code dictionary then?

[the remark this answers is not in the document]
Comment (vikivivi) [reply codespell_lib/data/dictionary.txt:28284]: No. Do not move to dictionary_code.txt. This is fine.
code dictionary is meant for entry where `(valid_word_from_code)->(valid_word_from_English)`.
For example: `uint->unit`, `jupyter->Jupiter`

**Review by vikivivi — APPROVED** [review]
LGTM

**Pull request merged**